### DISCUSSION TITLE
Adding and removing users from the room using redis

### DISCUSSION THREAD

**Discussion by tapanparmarcdx** [miguelgrinberg/Flask-SocketIO#2026] [General]
Hello,

I have following configuration with flask app. when I run 1 instance of the app it works fine.

I am running two gunicorn instances one on port 5001 and one on port 5002. when I connect client to port 5001 instance and then make a post `/items/<id>` request to update the item on instance running on port 5002 then it says no participants found in the room. If does send the message to them but I can't get participants back.  I am assuming that since I am using redis client manager I can get room participants from any instance. I also tried adding socketio channel while creating server but it didn't help.

I am trying to get participants to check their permissions again before sending updates.

Is there any other way to achieve this?

`__init__.py`
```
def create_app():
    app = Flask(__name__)
    app.register_blueprint(item_bp)
    return app

def create_websocket_app(app):
    redis_client_manager = RedisManager(url="redis://localhost:6379/0")
    
    socketio = SocketIO(
        app,
        client_manager=redis_client_manager,
        ping_interval=30,
        ping_timeout=30,
        resource="ws",
        async_mode="gevent",
    )
    
    socketio.on_event("connect", on_connect)
    socketio.on_event("subscribe_to_item", subscribe_to_item)
    
    return socketio

def on_connect(auth):
    # verify token

    sid = request.sid
    socketio = app.extensions["socketio"]

    socketio.server.save_session(sid, {"user_id": str(g.user_obj.id), "user": g.user_obj})


def subscribe_to_item(data):
    join_room(f"item_{data.item_id}")
```

`blueprint.py`
```
from flask import Blueprint, current_app as app

item_bp = Blueprint("item", __name__)

@item_bp.route("/items/<id>", methods=["POST"])
def update_item(id):
    # update Item
    socketio = app.extensions["socketio"]
    client_manager = socketio.server.manager
    clients = client_manager.get_participants("/", f"item_{id}")

    # Loop through client to check permission and remove those doesn't have permission any more
    socketio.send(update.to_json(), to=f"item_{id}", namespace="/")
```

Thanks

**Comment by miguelgrinberg** [comment]
There are no public methods that return the participants in a room. This is intentional, there is currently no mechanism to obtain a complete list across multiple hosts.

My recommendation is that you track room participants in a database, something this package has no access to.

**Reply by tapanparmarcdx** [reply]
Okay, Thank you
